shell/platform/tizen/channels/key_event_channel.cc

```diff
@@ -297,15 +146,110 @@ void KeyEventChannel::SendKey(const char* key,
   } else {
     event.AddMember(kTypeKey, kKeyUp, allocator);
   }
-  channel_->Send(event, [callback = std::move(callback)](const uint8_t* reply,
-                                                         size_t reply_size) {
-    if (reply != nullptr) {
-      std::unique_ptr<rapidjson::Document> decoded =
-          JsonMessageCodec::GetInstance().DecodeMessage(reply, reply_size);
-      bool handled = (*decoded)[kHandledKey].GetBool();
-      callback(handled);
+  channel_->Send(
+      event, [this, sequence_id](const uint8_t* reply, size_t reply_size) {
+        if (reply != nullptr) {
+          std::unique_ptr<rapidjson::Document> decoded =
+              JsonMessageCodec::GetInstance().DecodeMessage(reply, reply_size);
+          bool handled = (*decoded)[kHandledKey].GetBool();
+          ResolvePendingEvent(sequence_id, handled);
+        }
+      });
+}
+
+void KeyEventChannel::SendEmbedderEvent(const char* key,
+                                        const char* string,
+                                        const char* compose,
+                                        uint32_t modifiers,
+                                        uint32_t scan_code,
+                                        bool is_down,
+                                        uint64_t sequence_id) {
+  uint64_t physical_key = GetPhysicalKey(scan_code);
```

Comment:
`GetPhysicalKey`'s argument type is `int`. and `scan_code` type is `uint32_t`.  
Can you change the type to be the same?
And the guide says to avoid unsigned types. If this also applies to this code, I think it is better to change it.

https://google.github.io/styleguide/cppguide.html#Integer_Types

Reply:
> `GetPhysicalKey`'s argument type is `int`. and `scan_code` type is `uint32_t`.

Nice catch!

> And the guide says to avoid unsigned types. If this also applies to this code, I think it is better to change it.

Flutter key codes are assumed to be 52-bit unsigned integers. We should always represent them with `uint64_t`.